src/core/src/op/convert_align_types.cpp

```diff
@@ -0,0 +1,171 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/convert_align_types.hpp"
+
+#include "itt.hpp"
+#include "validation_util.hpp"
+
+namespace ov {
+namespace op {
+namespace {
+
+std::unordered_map<size_t, element::Type> bit_to_int{
+    {4, element::i4},
+    {8, element::i8},
+    {16, element::i16},
+    {32, element::i32},
+    {64, element::i64},
+};
+
+element::Type infer_types(const v14::ConvertAlignTypes* op) {
+    const auto lhs = op->input(0);
```

Comment:
Getting input is probably not required and `get_input_element_type` can be used instead to get input types.